pallets/ema-oracle/src/lib.rs

```diff
@@ -149,10 +152,17 @@ pub mod pallet {
 	pub enum Error<T> {
 		TooManyUniqueEntries,
 		OnTradeValueZero,
+		OracleNotFound,
 	}
 
 	#[pallet::event]
-	pub enum Event<T: Config> {}
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Oracle was added to the whitelist.
+		OracleAdded { source: Source, assets: (AssetId, AssetId) },
```

Comment:
when you add event like this, i would expect it being fired every time oracle is created, not just when you modify whitelist